client/post-editor/media-modal/index.jsx

```diff
@@ -241,9 +243,28 @@ export class EditorMediaModal extends Component {
 		this.props.setView( ModalViews.DETAIL );
 	};
 
-	onImageEditorCancel = imageEditorProps => {
-		const { mediaLibrarySelectedItems } = this.props;
+	handleUpdatePoster = ( { ID, posterUrl } ) => {
+		const { site } = this.props;
+
+		// Photon does not support URLs with a querystring component.
+		posterUrl = posterUrl && posterUrl.split( '?' )[ 0 ];
```

Comment:
we definitely want to avoid redefining the input arguments. it would be much less risky if we went ahead and created a new variable here instead, even if it seems a bit verbose

```js
const urlBeforeQuery = ( posterUrl || '' ).split( '?' )[ 0 ];

// or

const urlBeforeQuery = posterUrl.replace( /\?.+$/, '' );
```

what else could `posterUrl` be if not a string?